SpatialGDK/Source/SpatialGDK/Private/Tests/LoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,358 @@
+	// This is never emptied, so does not need an additional check for cleared fields
+	uint32 KVPairCount = Schema_GetObjectCount(ComponentObject, 2);
+	for (uint32 i = 0; i < KVPairCount; i++)
+	{
+		Schema_Object* KVPairObject = Schema_IndexObject(ComponentObject, 2, i);
+		uint32 Key = Schema_GetUint32(KVPairObject, SCHEMA_MAP_KEY_FIELD_ID);
+		WorkerRequirementSet Value = SpatialGDK::GetWorkerRequirementSetFromSchema(KVPairObject, SCHEMA_MAP_VALUE_FIELD_ID);
+
+		AclMap.Add(Key, Value);
+	}
+
+	return AclMap;
+}
+
+bool AclMapDelegatesComponents(const WriteAclMap& AclMap, const WorkerRequirementSet DelegatedRequirementSet,
+							   const TArray<Worker_ComponentId> DelegatedComponents)
+{
+	for (Worker_ComponentId ComponentId : DelegatedComponents)
+	{
+		const auto Entry = AclMap.Find(ComponentId);
+		if (Entry == nullptr)
+		{
+			return false;
+		}
+		if (*Entry != DelegatedRequirementSet)
+		{
+			return false;
+		}
+	}
+	return true;
+}
+
+} // anonymous namespace
+
+LOADBALANCEENFORCER_TEST(GIVEN_a_view_with_no_data_WHEN_advance_load_balance_enforcer_THEN_return_no_acl_assignment_requests)
+{
+	SpatialGDK::FDispatcher Dispatcher;
+	SpatialGDK::EntityView View;
+	const TUniquePtr<SpatialVirtualWorkerTranslator> VirtualWorkerTranslator = CreateVirtualWorkerTranslator();
+	SpatialGDK::FSubView SubView(SpatialConstants::LB_TAG_COMPONENT_ID, SpatialGDK::FSubView::NoFilter, &View, Dispatcher,
+								 SpatialGDK::FSubView::NoDispatcherCallbacks);
+
+	bool bInvoked = false;
+
+	SpatialGDK::SpatialLoadBalanceEnforcer LoadBalanceEnforcer = SpatialGDK::SpatialLoadBalanceEnforcer(
+		ThisWorker, SubView, VirtualWorkerTranslator.Get(), [&bInvoked](SpatialGDK::EntityComponentUpdate) {
+			bInvoked = true;
+		});
+
+	// The view has no entities in it. We expect the enforcer not to produce any ACL requests.
+	LoadBalanceEnforcer.Advance();
+	TestTrue("LoadBalanceEnforcer did not try to send an ACL update", !bInvoked);
+
+	return true;
+}
+
+LOADBALANCEENFORCER_TEST(
+	GIVEN_load_balance_enforcer_with_valid_mapping_WHEN_asked_for_acl_assignments_THEN_return_correct_acl_assignment_requests)
+{
+	SpatialGDK::FDispatcher Dispatcher;
+	SpatialGDK::EntityView View;
+	const TUniquePtr<SpatialVirtualWorkerTranslator> VirtualWorkerTranslator = CreateVirtualWorkerTranslator();
+
+	AddLBEntityToView(View, EntityIdOne, ThisRequirementSet, OtherVirtualWorker);
+
+	SpatialGDK::FSubView SubView(SpatialConstants::LB_TAG_COMPONENT_ID, SpatialGDK::FSubView::NoFilter, &View, Dispatcher,
+								 SpatialGDK::FSubView::NoDispatcherCallbacks);
+
+	TArray<SpatialGDK::EntityComponentUpdate> Updates;
+
+	SpatialGDK::SpatialLoadBalanceEnforcer LoadBalanceEnforcer = SpatialGDK::SpatialLoadBalanceEnforcer(
+		ThisWorker, SubView, VirtualWorkerTranslator.Get(), [&Updates](SpatialGDK::EntityComponentUpdate Update) {
+			Updates.Add(MoveTemp(Update));
+		});
+
+	LoadBalanceEnforcer.Advance();
```

Comment:
Interested to know why we need to advance here?

Reply:
good point- we don't. I just used similar boilerplate across all of them. Will remove